src/v/cluster/controller_backend.cc

```diff
@@ -663,7 +663,9 @@ void controller_backend::process_delta(const topic_table::ntp_delta& d) {
     // update partition_leaders_table if needed
 
     if (d.type == topic_table_ntp_delta_type::removed) {
-        _partition_leaders_table.local().remove_leader(d.ntp, d.revision);
+        ssx::spawn_with_gate(_gate, [this, ntp = d.ntp, rev = d.revision] {
+            return _partition_leaders_table.local().remove_leader(ntp, rev);
+        });
```

Comment:
It's probably out of scope, but I'm still curious. We delete leaders here but add them via metadata_dissemination_service. How do we make sure these don't race so we don't end up with garbage leaders sitting in partition leaders table forever?